### DISCUSSION TITLE
Feature request. Use it with command in visual mode.

### DISCUSSION THREAD

**Discussion by aca** [linrongbin16/gitlinker.nvim#38] [Q&A] [answered]
Hi, thanks for maintaining repository.

I want this GBrowse command to show visual mode but it just doesn't work.
It only returns address with current line.

```
vim.api.nvim_create_user_command("GBrowse", function()
    require("gitlinker").link({action = require("gitlinker.actions").system})
end, {
    range = true,
})
```

If I choosed visual blocked L1-3, It returns 

```
35145fe0f/README#L1
```
not 
```
35145fe0f/README#L1-3
```

**Comment by linrongbin16** [comment]
Hi @aca , sorry for missing your message, visual mode should be support already, can you show me how you defined key mappings?

**Comment by linrongbin16** [comment]
BTW, you don't even need to define the `GBrowse` command, just use `<leader>gl` will give you visual selected lines.

**Comment by aca** [comment]
Thanks for the reply.
I have quite lot of mapping. I don't want to add another mapping that I might not use it every hour.

**Reply by linrongbin16** [reply]
That's fine, let's focus on the `GBrowse` command you have, can you tell me more details about how will you operate to use this `GBrowse` command?

For example, first type `V` to enter line-wise visual mode, then type `jjj` to move down 3 lines, then type `:GBrowse`, then type `<CR>` (the `ENTER` key) to execute ?

**Reply by aca** [reply]
Yap. I once did make it open git link but couldn't find a way to pass multi line. It only returns current line.

**Reply by linrongbin16** [reply]
in my knowledge, when you manually type `:` vim will quit visual mode, then you will lose what you have visual selected.

that is why I recommend you use a key mapping instead of manually type a command

**Reply by linrongbin16** [reply]
let me do some investigation see if it's supported

**Reply by linrongbin16** [reply]
Hi @aca , it's fixed in #57 , please update plugin and verify.
BTW, thanks for your report.

**Reply by linrongbin16** [reply]
The issue introduced again in #58 , let's track it in #47 .

**Reply by linrongbin16** [reply, marked as answer]
Hi @aca , finally I support your command range mode in #60 , please read: https://github.com/linrongbin16/gitlinker.nvim#customization
